numba/cuda/cudaimpl.py

```diff
@@ -489,8 +489,14 @@ def ptx_fp16_binary(context, builder, sig, args):
 
 
 lower_fp16_binary(stubs.fp16.hadd, 'add')
+lower_fp16_binary(operator.add, 'add')
+lower_fp16_binary(operator.iadd, 'add')
```

Comment:
`iadd` is inplace add "+=". We need to add tests to for all inplace operators. Also, [ptx](https://docs.nvidia.com/cuda/parallel-thread-execution/index.html#half-precision-floating-point-instructions) doesn't support `iadd` opcode. We might need to change the result register as the lhs in `lower_fp16_binary` for this feature.

Reply:
In python, "inplace" operation `a += b` can mean `a = a + b` for immutable types. All number types are immutable so it is okay to just reuse the implementation for the non-inplace version.

Reply:
clarifying: `a = a + b`, the `+` can create a new object which is then bound to the variable `a` while the old object in `a` is not modified.

example:

```python
>>> a = "a"
>>> b = "b"
>>> c = a
>>> a += b
>>> c
'a'
>>> a
'ab'
```

Reply:
TIL, thanks for clarifying!

Reply:
@isVoid just to check, do you think this is now resolved? (https://github.com/numba/numba/pull/7885/commits/7b36527c847c28fe1abba228753b954f56db539b added some inplace binary tests - what do you think of them)?

Reply:
I've looked over the tests and have made comments there, so I'll resolve this conversation.